content/appendices.md

```diff
@@ -663,6 +663,9 @@ Room IDs are case-sensitive. They are not meant to be
 human-readable. They are intended to be treated as fully opaque strings
 by clients.
 
+The length of a room ID, including the `!` sigil and the domain, MUST
+NOT exceed 255 bytes.
```

Comment:
This was also previously specified in the C-S API, also going back to https://github.com/matrix-org/matrix-spec/commit/6a0595bc46c3751090d5d24e2ab94647c9573915.